prow/cmd/deck/job_history.go

```diff
@@ -385,6 +391,19 @@ func getBuildData(ctx context.Context, bucket storageBucket, dir string) (buildD
 	return b, nil
 }
 
+func getPullRequest(jobAgent *jobs.JobAgent, jobName, buildID string) (*prowv1.Pull, error) {
+	j, err := jobAgent.GetProwJob(jobName, buildID)
```

Comment:
For the bigger Prow instances like k8s and openshift, this will only work for about 24 hours, after which the prowjobs are deleted from the kubernetes cluster. The purpose of the view you are changing is to view historical data from GCS, which means that 99% of the jobs you are seeing there won't be in the Kubernetes cluster anymore and thus this will not work.

Reply:
Thanks Alvaro, I wasn't aware of this limitation.
I changed the implementation to read `prowjob.json` from GCS instead of using JobAgent cache.